src/Mvc/Mvc.Core/src/JsonOptions.cs

```diff
@@ -42,7 +43,7 @@ public class JsonOptions
         // The JsonSerializerOptions.GetTypeInfo method is called directly and needs a defined resolver
         // setting the default resolver (reflection-based) but the user can overwrite it directly or calling
         // .AddContext<TContext>()
-        TypeInfoResolver = CreateDefaultTypeResolver()
+        TypeInfoResolver = TrimmingAppContextSwitches.EnsureJsonTrimmability ? null : CreateDefaultTypeResolver()
     };
 
     private static IJsonTypeInfoResolver CreateDefaultTypeResolver()
```

Comment:
Doesn't this need a suppression?

Reply:
Oh yes and no, MVC is not marked as Trimmable yet, so, we are not getting error. However, let me add it.